### PR TITLE
AliasAnalysis handles SqueezeOp.

### PR DESCRIPTION
**Pull request by wujingyue**

### CODE REVIEW

**Comment by wujingyue** [comment]
!build

File: csrc/alias_analysis.cpp [hunk 1]
```diff
@@ -332,6 +333,40 @@ void AliasFinder::handle(const SliceOp* slice) {
...
+  std::unordered_map<IterDomain*, IterDomain*> in_rfactor_to_out_root =
+      PairwiseRootDomainMap(in, out).mapBroadcast(true).mapProducerToConsumer();
+
+  Layout out_layout;
+  // Preserve the allocation order of existing dimensions.
+  for (const auto i : c10::irange(in_layout.allocation_domain.size())) {
+    IterDomain* in_allocation_id = in_layout.allocation_domain[i];
+    if (in_rfactor_to_out_root.count(in_allocation_id) == 0) {
+      continue;
+    }
```
Comment (jacobhinkle) [inline csrc/alias_analysis.cpp:361]: You could potentially add a check for `isReduction()` here then factor this out as a generic way to preserve allocation domain of the input, in conjunction with #1583 .

**Review by jacobhinkle — APPROVED** [review]
LGTM

File: csrc/alias_analysis.cpp (hunk 1 above)
Comment (wujingyue) [reply csrc/alias_analysis.cpp:401]: Great idea -- I'm indeed tired of writing the same pattern over and over again. Will do that in my next PR.

**Comment by wujingyue** [comment]
!build

**Pull request merged**